client/src/pages/CreateOrganisationDetails/SecondStep.js

```diff
@@ -0,0 +1,197 @@
+import { useReducer, useEffect, useRef } from 'react';
+import { useHistory } from 'react-router-dom';
+import { useMediaQuery } from 'react-responsive';
+import { breakpoints } from '../../theme';
+
+import { Grid, Typography as T, Inputs as I, Button } from '../../components';
+import * as S from './style';
+import { createOrganisationDetails as validate } from '../../validation/schemas';
+import { Organisations } from '../../api-calls';
+
+import { navRoutes as R } from '../../constants';
+import { useAuth } from '../../context/auth';
+
+const { Row, Col } = Grid;
+
+const initialState = {
+  mainColor: '',
+  secondaryColor: '',
+  neutralColor: '',
+  httpError: '',
+  validationErrs: {},
+  loading: false,
+  displayColorPicker: false,
+};
+
+function reducer(state, newState) {
+  let value = newState;
+  if (typeof newState === 'function') {
+    value = newState(state);
+  }
+
+  return { ...state, ...value };
+}
+
+const SecondStep = () => {
+  const { user } = useAuth();
+  const submitAttempt = useRef(false);
+  const [state, setState] = useReducer(reducer, initialState);
+  const {
+    mainColor,
+    secondaryColor,
+    neutralColor,
+    loading,
+    validationErrs,
+    httpError,
+  } = state;
+
+  const history = useHistory();
+
+  const isMobile = useMediaQuery({
+    query: `(max-width: ${breakpoints.mobile})`,
+  });
+
+  useEffect(() => {
+    if (submitAttempt.current) {
+      validateForm();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [mainColor, secondaryColor, neutralColor]);
+
+  const validateForm = () => {
+    try {
+      validate({
+        secondStep: true,
+        mainColor,
+        secondaryColor,
+        neutralColor,
+      });
+      setState({ validationErrs: {} });
+      return true;
+    } catch (error) {
+      if (error.name === 'ValidationError') {
+        setState({ validationErrs: { ...error.inner } });
+      }
+      return false;
+    }
+  };
+
+  const handleUpdate = async () => {
+    setState({ loading: true });
+    const { error } = await Organisations.updateOrganisation({
+      id: 1,
+      body: {
+        colors: {
+          main: mainColor,
+          secondary: secondaryColor,
+          neutral: neutralColor,
+        },
+      },
+    });
+
+    setState({ loading: false });
+    if (error) {
+      if (error.statusCode === 409) {
+        setState({ validationErrs: { email: error.message } });
+      } else {
+        setState({ httpError: error.message });
+      }
+    } else {
+      // after that the user should be directed to its dashboard
+      history.push(R.ADMIN.CREATE_UNIQUE_LINK);
+    }
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+
+    submitAttempt.current = true;
+
+    const isValid = validateForm();
+    if (isValid) {
+      handleUpdate();
+    }
+  };
+
+  return (
+    <S.Form onSubmit={handleSubmit}>
+      <T.H1 weight="bold">Let’s customise!</T.H1>
```

Comment:
Put into a row or add ml to being it in line with the elements below it

Reply:
done